src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractMask.java

```diff
@@ -63,6 +63,9 @@
 
     @Override
     public boolean laneIsSet(int i) {
+        if (i < 0 || i >= length()) {
+            throw new IllegalArgumentException("Index " + i + " must be zero or positive, and less than " + length());
+        }
```

Comment:
Apologies i should have previously said use `Objects.checkIndex(i, length())`. In fact lets call `length` once:
```
    int length = length();
    Objects.checkIndex(i, length);
    if (length <= Long.SIZE) {
    ...
```

Separately i wonder if the second bounds check should be `if i <= Long.SIZE`, since the first bounds check should dominate in many cases?

(I know there are other areas where we should `Objects.checkIndex` e.g. `*MaxVector.lane/withLane` but we can fix those later.)

Reply:
With length <= Long.SIZE there is a better possibility of optimizing out else part and thus only fast path gets generated. length is final static values and hence in most of the cases due to in-lining this constant may get propagated.